From 17cee31c57b4798d936987d66ac2d4510b4707ce Mon Sep 17 00:00:00 2001
From: Tom Birdsong <tbirdsong@nvidia.com>
Date: Tue, 14 Apr 2026 06:56:26 -0700
Subject: [PATCH] build: install MatX for Holoscan packaging

Holoscan embeds MatX while producing an SDK development package. Enable
the dependency install rules for the nested build and place its headers
under the SDK third-party include layout.

Upstream-Status: Inappropriate [holoscan specific]

diff --git a/CMakeLists.txt b/CMakeLists.txt
index dc8ef29f..8eea93ad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -426,13 +426,13 @@ target_link_libraries(matx INTERFACE CUDA::nvtx3)
 
 # Build config files if the user isn't adding this as a subdirectory. At this point our transitive target
 # should have all build properties needed based on the options passed in
-if (NOT_SUBPROJECT)
+if (ON)
     include(GNUInstallDirs)
     include(CMakePackageConfigHelpers)
 
     install(TARGETS matx EXPORT matx-exports)
-    install(DIRECTORY include/ DESTINATION include)
-    install(FILES ${CMAKE_BINARY_DIR}/include/matx/version_config.h DESTINATION include)
+    install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/3rdparty/matx)
+    install(FILES ${MATX_BINARY_DIR}/include/matx/version_config.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/3rdparty/matx)
 
     set(doc_string
     [=[
